File: backend/src/main/java/com/grupo6daw/lcdd_daw/dto/UserDTO.java
package com.grupo6daw.lcdd_daw.dto;

import java.util.List;

public record UserDTO(
        Long userId,
        String userName,
        String userSurname,
        String userNickname,
        String userInterests,
        String userEmail,
        ImageDTO userImage,
        List<String> userRoles,
        List<EventBasicDTO> userRegisteredEvents,
        List<EventBasicDTO> userOwnEvents,
        List<GameDTO> userFavGames,
        List<NewBasicDTO> userNews) {
}

File: frontend/app/dtos/UserDTO.ts
import type EventBasicDTO from "./EventBasicDTO";
import type GameDTO from "./GameDTO";
import type ImageDTO from "./ImageDTO";
import type NewBasicDTO from "./NewBasicDTO";

export default interface UserBasicDTO {
    userId: number,
    userName: string,
    userSurname: string,
    userNickname: string,
    userInterests: string,
    userEmail: string,
    userImage: ImageDTO,
    userRoles: Array<String>,
    userRegisteredEvents: Array<EventBasicDTO>,
    userOwnEvents: Array<EventBasicDTO>,
    userFavGames: Array<GameDTO>,
    userNews: Array<NewBasicDTO>
}